lib/log_decorator/logging.rb

```diff
@@ -0,0 +1,33 @@
+module LogDecorator
+  module Logging
+    module ClassMethods
+      def instance_logger
+        @instance_logger ||= LogProxy.new(name, '#')
+      end
+
+      def _log
+        @_log ||= LogProxy.new(name, '.')
+      end
+
+      def _logger=(logger)
+        _log.logger = logger
+        instance_logger.prefix = logger
+      end
+
+      def _prefix=(prefix)
```

Comment:
I'm thinking this should be `_log_prefix`, because prefix seems way too generic (even with the `_` at the beginning).  Additionally, because it's generic, it reads funny on the caller's side, when read out of context (e.g. `TestClass1._prefix = "MIQ"`)

The internal `prefix` method on the LogProxy is fine though because it's scoped to the LogProxy.